fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashViewModel.java

```diff
@@ -41,412 +42,473 @@
-        return getLao(getLaoId().getValue());
-    }
-
-    @Nullable
-    public Set<PublicKey> getAttendeesFromTheRollCall() throws NoRollCallException {
-        return getCurrentLao().lastRollCallClosed().getAttendees();
-    }
-
-    @Nullable
-    public PublicKey getOrganizer() {
-        return getCurrentLao().getOrganizer();
-    }
-
-    @Nullable
-    public List<String> getAttendeesFromTheRollCallList() throws NoRollCallException {
-        List<String> list = new ArrayList<>();
-        Iterator<PublicKey> pub = Objects.requireNonNull(getAttendeesFromTheRollCall()).iterator();
-        while (pub.hasNext()) {
-            String current = pub.next().getEncoded();
-            list.add(current);
-        }
-        return list;
-    }
-
-    @Nullable
-    private Lao getLao(String laoId) {
-        LAOState laoState = laoRepository.getLaoById().get(laoId);
-        if (laoState == null) return null;
-        return laoState.getLao();
-    }
-
-    public boolean canPerformTransaction(
-            String currentAmount, String currentPublicKeySelected, int radioGroup) {
-        if ((currentAmount.isEmpty()) || (Integer.parseInt(currentAmount) < MIN_LAO_COIN)) {
-            // create in View Model a function that toast : please enter amount
-            requireToPutAnAmount();
-            return false;
-        } else if (currentPublicKeySelected.isEmpty() && (radioGroup == NOTHING_SELECTED)) {
-            // create in View Model a function that toast : please enter key
-            requireToPutLAOMember();
-            return false;
-        } else {
-            return true;
-        }
-    }
-
-    private void processNotCoinbaseTransaction(PrivateKey privK, PublicKey pubK, List<Output> outputs, long amountFromReceiver, List<Input> inputs) throws GeneralSecurityException {
-        int index;
-        String transactionHash;
-        List<TransactionObject> transactions =
-                getCurrentLao().getTransactionByUser().get(pubK);
-
-        long amountSender =
-                TransactionObject.getMiniLaoPerReceiverSetTransaction(
-                        transactions, pubK)
-                        - amountFromReceiver;
-        Output outputSender =
-                new Output(amountSender, new ScriptOutput(TYPE, pubK.computeHash()));
-        outputs.add(outputSender);
-        for (TransactionObject transactionPrevious : transactions) {
-            transactionHash = transactionPrevious.computeTransactionId();
-            index = transactionPrevious.getIndexTransaction(pubK);
-            inputs.add(processSignInput(privK, pubK, outputs, transactionHash, index));
-        }
-    }
+  public static final String TAG = DigitalCashViewModel.class.getSimpleName();
+  private static final String LAO_FAILURE_MESSAGE = "failed to retrieve lao";
+  private static final String PUBLISH_MESSAGE = "sending publish message";
+  private static final String RECEIVER_KEY_ERROR = "Error on the receiver s public key";
```

Comment:
Looks like other string for error/notifications are in string.xml, don't know if it is worth to move them ?

Reply:
This is from master. Moreover this is not displayed which is what the strings.xml is usually for